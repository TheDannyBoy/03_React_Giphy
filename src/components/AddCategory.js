import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = evt => {
    setInputValue(evt.target.value);
  };

  const handleSubmit = evt => {
    evt.preventDefault();
    setCategories(categories => ([ inputValue, ...categories ]));
    setInputValue('');
  };

  return (<form onSubmit={handleSubmit}>
    <h1>{ inputValue }</h1>
    <input
      type='text'
      value={inputValue}
      onChange={handleInputChange}>
    </input>
  </form>)
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
}
