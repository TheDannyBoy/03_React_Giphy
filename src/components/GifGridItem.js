import React from 'react'

export const GifGridItem = ({ title, url }) => {
  const classes = ['card'];
  return (
    <div className={classes.join(' ')}>
      <img src={url} alt={title} />
      <p>{ title }</p>
    </div>
  )
}
