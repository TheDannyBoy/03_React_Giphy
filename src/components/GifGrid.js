import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import getGifs from '../helpers/getGifs'

export const GifGrid = ({ category }) => {
  useEffect(() => {
    getGifs(category).then(setImagesList);
  }, [category]);

  const [imagesList, setImagesList] = useState([]);

  return (
    <>
      <h3>{ category }</h3>
      <div className='card-grid'>
        {
          imagesList.map(image => (
            <GifGridItem key={image.id} {...image} />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string,
}
