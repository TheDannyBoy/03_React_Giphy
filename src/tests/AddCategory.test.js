import { shallow } from "enzyme";
import { AddCategory } from "../components/AddCategory";

describe('Testing - AddCategory', () => {

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Should create AddCatefory correctly', () => {
    const setCategories = jest.fn();
    const wrapper = shallow(<AddCategory setCategories={setCategories} />)

    expect(wrapper).toMatchSnapshot();
  })

  test('Should submit changed data', () => {
    const setCategories = jest.fn();
    const wrapper = shallow(<AddCategory setCategories={setCategories} />)
    const input = wrapper.find('input');
    input.simulate('change', { target: { value: 'foo' } });
    const form = wrapper.find('form');
    form.simulate('submit', { preventDefault: jest.fn() });

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(input.prop('value')).toBe('');
  })
});
