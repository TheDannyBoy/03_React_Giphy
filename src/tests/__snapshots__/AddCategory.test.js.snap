// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing - AddCategory Should create AddCatefory correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h1 />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
